packages/saltcorn-admin-models/models/backup.ts

```diff
@@ -444,31 +445,57 @@ function executableIsAvailable(name: string) {
  * @param {string} dir
  * @returns {Promise<void>}
  */
-const extract = async (fnm: string, dir: string): Promise<void> => {
-  const backup_with_system_zip = executableIsAvailable("unzip");
+const extract = async (
+  fnm: string,
+  dir: string,
+  password?: string
+): Promise<void> => {
   const state = getState();
+  const backup_with_system_zip = executableIsAvailable("unzip");
+
   if (backup_with_system_zip) {
     return await new Promise((resolve, reject) => {
-      var subprocess = spawn("unzip", [File.normalise(fnm), "-d", dir]);
+      const args = [
+        ...(password ? [`-P${password}`] : []),
+        File.normalise(fnm),
+        "-d",
+        dir
+      ];
+      
+      const subprocess = spawn("unzip", args);
+      
       subprocess.stdout.on("data", (data: any) => {
         state.log(6, data.toString());
       });
+      
       subprocess.stderr.on("data", (data: any) => {
-        state.log(1, data.toString());
-      });
-      subprocess.on("close", function (exitCode: any) {
-        if (exitCode != 0) reject(new Error("unzip failed"));
-        else resolve(undefined);
+        const output = data.toString();
+        state.log(1, output);
+        if (output.includes("password") || output.includes("encrypted")) {
```

Comment:
please run prettier. i have prettier 3.3.3 installed - you can use through VS code